bioblend/galaxyclient.py

```diff
@@ -90,21 +106,23 @@ def make_post_request(self, url, payload, params=None, files_attached=False):
             params = self.default_params
 
         if files_attached:
-            payload.update(params)  # merge query string values into request body instead
-            poster.streaminghttp.register_openers()
-            datagen, headers = poster.encode.multipart_encode(payload)
-            request = Request(url, datagen, headers)
-            fp = urlopen(request)
-            return json.loads(fp.read())
+            payload.update(params)
+            payload = MultipartEncoder(fields=payload)
+            headers = self.json_headers.copy()
+            headers['Content-Type'] = payload.content_type
+            params = {}
         else:
             payload = json.dumps(payload)
-            r = requests.post(url, data=payload, headers=self.json_headers,
-                              verify=self.verify, params=params)
-            if r.status_code == 200:
-                return r.json()
-            # @see self.body for HTTP response body
-            raise ConnectionError("Unexpected response from galaxy: %s" %
-                                  r.status_code, body=r.text)
+            headers = self.json_headers
+            params = params
```

Comment:
Disagree that it should be removed but I have refactored it in ef2e903 as a compromise.